tests/test_pylint_runners.py

```diff
@@ -57,6 +59,25 @@ def test_runner_with_arguments(runner: _RunCallable, tmpdir: LocalPath) -> None:
         assert err.value.code == 0
 
 
+def test_pylint_argument_deduplication(tmpdir: LocalPath) -> None:
+    """Check that the Pylint runner does not overreport on duplicate arguments."""
+    filepath = os.path.abspath(__file__)
+    # Ensure that we have 3 branches in this file.
+    if filepath:
+        testargs = shlex.split("--report n --score n --max-branches 3")
```

Comment:
Maybe this is too fragile? If we're worried about other integration tests having more than 3 branches in this file, then I could try to find a different metric for overcounting (on duplication) or else spread the `testargs.extend` as `testargs.append` across four separate branches to increase the number of branches (and change `--max-branches` to match).

Reply:
Let's use the functional file for this (https://github.com/PyCQA/pylint/blob/main/tests/functional/t/too/too_many_branches.py), it would be expected that the new test break if we change the other file.

Reply:
Good idea - I should've thought of that. Was very tired when I wrote the integration test.
Changed.